artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonAbstractReceiver.java

```diff
@@ -78,6 +78,8 @@ public ProtonAbstractReceiver(AMQPSessionCallback sessionSPI,
       this.creditRunnable = createCreditRunnable(connection);
       useModified = this.connection.getProtocolManager().isUseModifiedForTransientDeliveryErrors();
       this.routingContext = new RoutingContextImpl(null).setDuplicateDetection(connection.getProtocolManager().isAmqpDuplicateDetection());
+      this.standardMessageReader = new AMQPMessageReader(this);
+      this.largeMessageReader = new AMQPLargeMessageReader(this, sessionSPI.isLargeMessgeSync());
```

Comment:
This change is unnecessary if you just use the API add in the session context to check the value